CMakeLists.txt

```diff
@@ -51,6 +51,14 @@ endif()
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 1)
 set(NVIM_VERSION_PATCH 1)
+# Package maintainers should define this according to their package system
+# conventions, if the source code or runtime files were modified, e.g.:
+#   cmake -DNVIM_VERSION_PRERELEASE=-deb3
+# or define it as empty to prevent "-dev", if no modifications were made:
+#   cmake -DNVIM_VERSION_PRERELEASE=
+if(NOT DEFINED NVIM_VERSION_PRERELEASE)
+  set(NVIM_VERSION_PRERELEASE "-dev")
+endif()
```

Comment:
@jamessan @fwalch Note regarding packaging: if you build in a git tree, then this may not affect you. If you don't, then the `-dev` suffix will be appended to the version string unless you define `NVIM_VERSION_PRERELEASE` to be empty.

This avoids having to physically remove `NVIM_VERSION_PRERELEASE` in our source (e.g. 4a1c36e953559382362b79be7630a64163c43ef2) every time we tag a release.